src/features/Apiexplorer/SubscribeRenderer/index.tsx

```diff
@@ -51,13 +57,13 @@ function SubscribeRenderer<T extends TSocketSubscribableEndpointNames>({
   };
 
   const handleClick = useCallback(() => {
-    unsubscribe();
-    subscribe(parseRequestJSON());
+    if (subscribe_ref.current) subscribe_ref.current.unsubscribe();
+    subscribe_ref.current = subscribe(parseRequestJSON());
```

Comment:
I see that you try to execute and write to a ref during rendering which I think might cause issues later on. I was wondering if it perhaps also just works with using useState instead of ref?

To give an idea, I had something like this in mind:

```
  const [is_subscribed, setIsSubscribed] = useState(false);

  useEffect(() => {
    const subscription = subscribe(parseRequestJSON());
    if (is_subscribed) {
      subscription();
    } else {
      subscription().unsubscribe();
    }
  }, [is_subscribed])

  const handleClick = () => setIsSubscribed(true);

  const handleClear = () => {
    setResponseState(false);
    setIsSubscribed(false);
  };
```

Reply:
We were using state before, the issue was whenever their was a rerender the state was reset that is why the issue was happening